code/mathematical_algorithms/src/steepest_descent/steepest_descent.cpp

```diff
@@ -33,57 +33,48 @@ double norm(const std::array<double, S> &x, const std::array<double, S> &y)
 }
 
 /*
-Method of Steepest Descent
+Adds up two matrix
 */
 template <long unsigned int S>
-std::array<double, S> steepestdescent(const std::array<std::array<double, S>, S> &A, const std::array<double, S> &b,
-    const double e = 0.0001, const double step = 0.1, const unsigned int IterMax = 10000)
+std::array<double, S> operator+(const std::array<double, S> &x, const std::array<double, S> &y)
 {
-    std::array<double, S> x = {0.0};
-    unsigned int k = 0;
+    std::array<double, S> result;
 
-    do
+    for (unsigned int i = 0; i < S; ++i)
```

Comment:
Why not long unsigned int?

Reply:
I don't know, only rearrange order of functions

Reply:
Ok I see.